Spanner/tests/Unit/Session/CacheSessionPoolTest.php

```diff
@@ -709,20 +708,19 @@ private function getDatabase($shouldCreateThrowException = false, $willDeleteSes
             ->willReturn(self::DATABASE_NAME);
 
         if ($shouldCreateThrowException) {
-            $database->createSession()
-                ->willThrow(new \Exception());
+            $connection->createSessionAsync()->willThrow(new \Exception());
         } else {
-            $database->createSession(Argument::any())
-                ->will(function ($args, $mock, $method) use ($createdSession) {
+            $test = $this;
```

Comment:
What's the need for this?

Reply:
We use it as part of the closure on line 715 below. We reference the test so we can create a session mock each called createSessionAsync method on line 721-723. Without passing in `$test`, the `$this` var refers to the ObjectProphecy which can't create any mocks 😞

Reply:
Was there an issue with the way it was? Create the mock, and pass it into the closure's scope?

Also, I don't think `createSessionAsync` will throw an exception directly or return the session - it should be a promise.

Reply:
Yeah, you're right, this is completely unnecessary! I had rewritten it this way in the first draft when I had the database creating all of the sessions and I needed to return multiple sessions from the same function! Thanks for the 👀 on this one.

Reply:
I'm double checking to see if there's any way an exception could be thrown. Request exceptions should be handled by Guzzle's promises.

Reply:
Okay, I looked through that and I don't see a reasonable way an exception would be thrown, all of the calls should return a rejected a promise instead. I updated the test to return a RejectedPromise instead of throwing an exception.

I double checked the behavior in CacheSessionPool and there was a line where it caught the exception, and if there was an exception, skipped a test in processing the data queue.

I believe the intent of the exception was to indicate that sessions were not created. I changed that behavior to look at the length of createdSessions since we return an empty result if the create sessions requests failed.

@dwsupplee and @jdpedrie - Does this logic look right to you? Can you think of any edge cases I may have missed? Thanks!

Reply:
Looks reasonable to me!